src/ast/Guest.ts

```diff
@@ -37,15 +38,24 @@ export default class Guest extends Node {
       throw new ParserError("expected a guest name", this.lineNumber);
     }
     this.email = token; 
-  } 
+  }
+
   evaluate(context: object[]): void {
-    throw new Error("Method not implemented.");
+    // throw new Error("Method not implemented.");
   }
-  nameCheck(): void {
-    throw new Error("Method not implemented.");
+
+  nameCheck(map: object): void {
+    if (!map.hasOwnProperty(this.id)) {
+      let values = [null, this.name,this.email]; 
+      Object.assign(map, {[this.id ]: values});
+    } else {
+      throw new NameCheckError(`Guest with identifier ${this.id} is already defined.`); 
```

Comment:
Do we have the line numbers to add more context to the NameCheckError messages too?